### DISCUSSION TITLE
Order Book Snapshots and Aggregated Market Depth

### DISCUSSION THREAD

**Discussion by BGTCapital** [bmoscon/cryptofeed#746] [Q&A]
Hi
We are trying to use cryptofeed to returns metrics on average the depth of the order book (the cummulative volume of the base asset at 0.1%, 0.2%, 0.3%, 0.4%, 0.5%, 0.6%, 0.7%, 0.8%, 0.9%, 1%, 1.5%, 2%, 4%, 6%, 8% and 10% from the mid price) per snapshot. For each interval, the aggregates are calculated by taking the average metrics of each snapshot within that interval. For example, the aggregated 1 hour spread is calculated by taking all spreads of each snapshot within an hour and calculating the average.

Could you kindly help us understand how can we do this at various depths and time slots (1m, 5min, 1H, 1D, etc)?
Thanks

**Comment by BGTCapital** [comment]
could you @bmoscon  kindly help us with this?. How can we collect cumulative volume of the base asset at 0.1%, 0.2%, 0.3%... from mid price per snapshot?

**Reply by bmoscon** [reply]
thats not really related to cryptofeed, so sorry, I cannot.

**Comment by peedrr** [comment]
Take a look at TimescaleDB. It's a time-series database, written as an extension of PostgreSQL. Cryptofeed's Postgres backend can feed Timescale and from there you can fairly easily use their Continuous Aggregates functions to formulate this data
